src/sentry/api/validators/monitor.py

```diff
@@ -145,3 +145,4 @@ class MonitorCheckInValidator(serializers.Serializer):
         )
     )
     duration = EmptyIntegerField(required=False, allow_null=True)
+    environment = serializers.CharField(required=False, allow_null=True)
```

Comment:
Can we explicitly disallow this to be changed in the checkin update endpoint?

Right now I think you could pass it and no matter what you pass it will be a no-op right.

Reply:
Changed how? I think we can just accept whatever we get and then `get_or_create()` the environment

Reply:
Right now this is just undocumented...we're just not using `MonitorEnvironment` anywhere

Reply:
Like in the checkin update endpoint, we should explicitly check if they're trying to change the environment of the checkin. That should not be allowed and we should error on it. Right now it would just do nothing